infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationAckReader.java

```diff
@@ -98,28 +98,51 @@ private long getMaxReplicatedStreamsTail() {
      *
      * For Log Entry Sync, this function returns the total number of entries remaining to be sent across all replicated
      * streams.
-     * For Snapshot Sync, each entry sent is a snapshot.  So this function returns the total number of snapshots
-     * remaining to be sent.
-     * If the ack'd timestamp is uninitialized, it returns 100%, which means no replication has been done.
+     *
+     * For Snapshot Sync, each entry sent is a snapshot of the corresponding stream.  The stream could have been
+     * checkpointed and trimmed so we cannot find the remaining number of entries accurately.  In this case, we simply
+     * subtract the acked timestamp from the global log tail when the snapshot sync started.
+     * Note that this method is not accurate because the global tail advances on writes for both replicated and
```

Comment:
This is true but a bit misleading as someone could say well don't return global tail but maxTailStreamsToReplicate, and it still holds true that this is not accurate as there is no guarantee that up to maxTailStreamsToReplicate all data is to be replicated. We might rephrase saying something like: Note this method is not accurate as the max tail can reflect the interleaving of replicated and non-replicated streams (or something similar)...